test/utils/test_convert.py

```diff
@@ -611,3 +612,231 @@ def test_from_dgl_hetero_graph():
+    G.add_edges_from([(0, 1, {'age': 2}), (1, 2, {'age': 7})])
+
+    data = from_hetero_networkx(G, node_type_attribute="type",
+                                edge_type_attribute=None)
+
+    assert data['A'].age.tolist() == [1, 6, 5]
+    assert data[('A', 'to', 'A')].edge_index.tolist() == [[0, 1, 1, 2],
+                                                          [1, 0, 2, 1]]
+    assert data[('A', 'to', 'A')].age.tolist() == [2, 2, 7, 7]
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_raise_missing_node_type_attribute():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node(0, type='A')
+    G.add_node(1)
+    G.add_edge(0, 1)
+
+    with pytest.raises(KeyError) as _:
+        from_hetero_networkx(G, node_type_attribute="type",
+                             edge_type_attribute=None)
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_raise_missing_edge_type_attribute():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node(0, type='A')
+    G.add_node(1, type='A')
+    G.add_edge(0, 1)
+
+    with pytest.raises(KeyError) as _:
+        from_hetero_networkx(G, node_type_attribute="type",
+                             edge_type_attribute="type")
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_raise_different_edge_attribute():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node(0, type='A')
+    G.add_node(1, type='A')
+    G.add_node(2, type='A')
+    G.add_edge(0, 1, a=1)
+    G.add_edge(0, 2, b=2)
+
+    with pytest.raises(ValueError) as _:
+        from_hetero_networkx(G, node_type_attribute="type",
+                             edge_type_attribute=None)
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_raise_different_node_attribute():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node(0, type='A', a=1)
+    G.add_node(1, type='A', b=1)
+
+    with pytest.raises(ValueError) as _:
+        from_hetero_networkx(G, node_type_attribute="type",
+                             edge_type_attribute=None)
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_works_with_named_edge_types():
+    import networkx as nx
+
+    G = nx.DiGraph()
+    G.add_node(0, type="A", x=1)
+    G.add_node(1, type="A", x=1)
+    G.add_node(2, type="B", y=1)
+    G.add_edge(0, 1, type=('A', 'towards', 'A'), x=1)
+    G.add_edge(0, 2, type=('A', 'towards', 'B'))
+
+    data = from_hetero_networkx(G, node_type_attribute="type",
+                                edge_type_attribute="type")
+
+    assert data[('A', 'towards', 'A')].edge_index.tolist() == [[0], [1]]
+    assert data[('A', 'towards', 'B')].edge_index.tolist() == [[0], [0]]
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_works_with_non_string_types():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node(0, type=1, x=1)
+    G.add_node(1, type=False, x=1)
+
+    data = from_hetero_networkx(G, node_type_attribute="type",
+                                edge_type_attribute=None)
```

Comment:
Mention in the docs that the `node_type_attribute` will be cast to a string, and raise an error if we can't cast to string.